plugin/plugin_manager.go

```diff
@@ -0,0 +1,189 @@
+package plugin
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	getter "github.com/hashicorp/go-getter"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/odpf/optimus/config"
+	"github.com/odpf/optimus/plugin/yaml"
+)
+
+var (
+	// for plugin installation, discovery & sync on both client and server
+	PluginsDir         = ".plugins"
+	PluginsArchiveName = "yaml-plugins.zip"
+)
+
+type IPluginManager interface {
+	Install(dst string, sources ...string) error
+	Archive(name string) error        // by server
+	UnArchive(src, dest string) error // by client
```

Comment:
If the interface is not consumed as a whole, should be broken down in 2 different interfaces and relevant modules should depend on the corresponding one.

Reply:
This interface is what the plugin manager library is exposing. the client that uses this manager can have restricted interfaces. For now I am keeping things simple.